File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.js
function solution(ingredient) {
    const table = [];
    let count = 0;
    
    for (let i of ingredient) {
        table.push(i);
        
        if (table.length < 4) continue;
        
        let baseIndex = table.length - 4;
        
        if (table[baseIndex] === 1
            && table[baseIndex + 1] === 2
            && table[baseIndex + 2] === 3
            && table[baseIndex + 3] === 1) {
            for (let i = 0; i < 4; i++) {
                table.pop();
            }
            
            count++;
        }
    }
    
    return count;
}

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    stack = []
    result = 0
    
    for x in ingredient:
        stack.append(x)
        
        if len(stack) < 4:
            continue
            
        if stack[-4] == 1 and stack[-3] == 2 and stack[-2] == 3 and stack[-1] == 1:
            for _ in range(4):
                stack.pop()
                
            result += 1
                
    return result
    